src/parser/view_transform.c

```diff
@@ -3189,6 +3189,13 @@ mq_copypush_sargable_terms_helper (PARSER_CONTEXT * parser, PT_NODE * statement,
 
   copy_cnt = -1;
 
+  /* check inst num in derived table. can not push predicate into subquery having inst num */
+  has_derived_table_inst = pt_has_inst_or_orderby_num_in_where (parser, new_query);
+  if (has_derived_table_inst)
+    {
+      return push_cnt;
```

Comment:
it looks better to return 0 for readablity.

Reply:
I fixed it.